src/Confluent.Kafka.Avro/GenericSerializerImpl.cs

```diff
@@ -31,9 +32,17 @@ internal class GenericSerializerImpl : IAvroSerializerImpl<GenericRecord>
         private int initialBufferSize;
         private bool isKey;
 
-        private Dictionary<Avro.RecordSchema, string> knownSchemas = new Dictionary<Avro.RecordSchema, string>();
-        private HashSet<KeyValuePair<string, string>> registeredSchemas = new HashSet<KeyValuePair<string, string>>();
-        private Dictionary<string, int> schemaIds = new Dictionary<string, int>();
+        private ConcurrentDictionary<Avro.RecordSchema, string> knownSchemas 
```

Comment:
Same discussion here as for the Deserializer, but it is worse here because there are three different dicts that should be in sync but the lack of common locking will lead to the opposite.

Reply:
the dicts don't need to always be in sync. there is a lock that covers the only case where this is important. 

but on reflection, i re-agree that preventing a spike in access to SR is important, particularly in the serializer case (going back to my initial implementation).